Client/Frontend/Browser/Playlist/Controllers/PlaylistFolderController.swift

```diff
@@ -0,0 +1,595 @@
+    }
+}
+
+extension PlaylistFolderController: UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return Section.allCases.count
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        guard let section = Section(rawValue: section) else {
+            return 0
+        }
+        
+        switch section {
+        case .savedItems: return 1
+        case .folders: return othersFRC.fetchedObjects?.count ?? 0
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier) else {
+            return UITableViewCell(style: .subtitle, reuseIdentifier: cellIdentifier)
+        }
+        
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        
+        guard let section = Section(rawValue: indexPath.section) else {
+            return
+        }
+        
+        let folderIcon = UIImage(systemName: "folder")?.template
+        
+        switch section {
+        case .savedItems:
+            let itemCount = savedFolder?.playlistItems?.count ?? 0
+            
+            cell.imageView?.image = folderIcon
+            cell.textLabel?.text = Strings.PlaylistFolders.playlistSavedFolderTitle
+            cell.detailTextLabel?.text = "\(itemCount) \(itemCount == 1 ? Strings.PlaylistFolders.playlistFolderSubtitleItemSingleCount : Strings.PlaylistFolders.playlistFolderSubtitleItemCount)"
+            cell.detailTextLabel?.textColor = .secondaryBraveLabel
+            cell.accessoryType = .disclosureIndicator
+            cell.selectionStyle = .none
+        case .folders:
+            guard let folder = othersFRC.fetchedObjects?[safe: indexPath.row] else {
+                return
+            }
+            
+            let itemCount = folder.playlistItems?.count ?? 0
+            
+            cell.imageView?.image = folderIcon
+            cell.textLabel?.text = folder.title
+            cell.detailTextLabel?.text = "\(itemCount) \(itemCount == 1 ? Strings.PlaylistFolders.playlistFolderSubtitleItemSingleCount : Strings.PlaylistFolders.playlistFolderSubtitleItemCount)"
+            cell.detailTextLabel?.textColor = .secondaryBraveLabel
+            cell.accessoryType = .disclosureIndicator
+            cell.selectionStyle = .none
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        52
+    }
+}
+
+extension PlaylistFolderController: UITableViewDelegate {
+    
+    private func deleteFolder(folder: PlaylistFolder) {
+        if PlaylistManager.shared.currentFolder?.objectID == folder.objectID {
+            PlaylistManager.shared.currentFolder = nil
+        }
+        
+        folder.playlistItems?.forEach({
```

Comment:
Do you think we should move the entire delete folder logic into PlaylistManager and take it our of the viewcontroller?